backend/lucene/src/main/java/org/hibernate/search/backend/lucene/scope/model/impl/LuceneDifferentNestedObjectCompatibilityChecker.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.scope.model.impl;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+
+import org.hibernate.search.backend.lucene.logging.impl.Log;
+import org.hibernate.search.util.common.logging.impl.LoggerFactory;
+
+public class LuceneDifferentNestedObjectCompatibilityChecker {
+
+	private static final Log log = LoggerFactory.make( Log.class, MethodHandles.lookup() );
+
+	private final LuceneScopeModel scopeModel;
+
+	private String absoluteFieldPath;
+	private List<String> nestedObjectPath;
+
+	public LuceneDifferentNestedObjectCompatibilityChecker(LuceneScopeModel scopeModel) {
+		this.scopeModel = scopeModel;
+	}
+
+	public void combineAndCheck(String anotherAbsoluteFieldPath) {
+		if ( absoluteFieldPath == null ) {
+			absoluteFieldPath = anotherAbsoluteFieldPath;
```

Comment:
I'm pretty sure introducing mutability here will come back and bite us... Can't you make it immutable, like the other checkers?

Reply:
Looks like this wasn't addressed?

Reply:
You're right. Doing now...